### ISSUE TITLE
Linux classroom assistant running but GUI doesn't appear

### ISSUE THREAD

**Issue by lizajane999** [education/classroom-assistant#97]
On Kubuntu Linux 18.04 when I start Classroom Assistant it seems to start, but the GUI does not appear. The process is running.

If I start from the command line, I get this:
```
$ classroom-assistant                                                                                                                      
(node:4144) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Command failed: xdg-settings set default-url-scheme-handler x-github-classroom classroom-assistant.desktop                                                                                                                                      
                                                                                                                                                                          
(node:4144) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
Then it just sits there.

**Comment by kevinwortman** [comment]
On plain Ubuntu Desktop 18.04 it runs and hangs, and I get

```
Gtk-Message: 22:26:19.460: Failed to load module "canberra-gtk-module"
(node:14062) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Command failed: xdg-settings set default-url-scheme-handler x-github-classroom classroom-assistant.desktop

(node:14062) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by kevinwortman** [comment]
And on Xubuntu 18.04 the message is

```
(node:14229) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Command failed: xdg-settings set default-url-scheme-handler x-github-classroom classroom-assistant.desktop
xdg-settings: default-url-scheme-handler not implemented for xfce

(node:14229) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.


```

**Comment by srinjoym** [comment]
@lizajane999 Just published a new build here which catches the error when the `xdg-settings` command fails:  https://github.com/education/classroom-assistant/releases/download/v1.0.2/classroom-assistant_1.0.2_amd64.deb

Another dependency for running Classroom Assistant on KDE is `gnome-keyring`, here are the installation instructions: https://github.com/education/classroom-assistant/blob/master/docs/linux-issues.md#kde

@kevinwortman The error handling should also take care of the issue you were hitting on Ubuntu 18.04, though I'm not sure why you were hitting that in the first place since this seems to be isolated to Kubuntu/other distros, let me know if the build above gets things working for you

We also do not support Xubuntu 18.04 as the `xdg-settings` command does not seem to be supported on that environment, feel free to open another issue about Xubuntu support, and we can discuss some manual installation steps!

**Comment by srinjoym** [comment]
Closing this out, looks like that build fixed the problem.

**Comment by kevinwortman** [comment]
This bug appears to be fixed in both plain Ubuntu 18.04, and also Xubuntu 18.04, in the new 1.0.2 package. Thanks!
